src/com/googlecode/jmxtrans/model/output/GangliaWriter.java

```diff
@@ -166,11 +169,14 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 			if (result.getValues() != null) {
 				for (final Map.Entry<String, Object> resultValue : result.getValues().entrySet()) {
 					final String name = KeyUtils.getKeyString(query, result, resultValue, getTypeNames());
-					final String value = resultValue.getValue().toString();
+					
+					BigDecimal value = new BigDecimal(resultValue.getValue().toString());
+					String strValue = value.compareTo(GANGLIA_MIN_DOUBLE) <= 0 ? "0" : value.toString();
+
 					GMetricType dataType = getType(resultValue.getValue());
-					log.debug("Sending Ganglia metric {}={} [type={}]", name, value, dataType);
+					log.debug("Sending Ganglia metric {}={} [type={}]", name, strValue, dataType);
```

Comment:
In case where we are modifying the data being sent, I think we should at least log an `info`, maybe a `warning`.